libafl_qemu/src/qemu/mod.rs

```diff
@@ -731,6 +731,39 @@ impl Qemu {
             .write_mem(addr, buf)
     }
 
+    /// Read a value from memory to a guest addr, taking into account the potential indirections with the current CPU.
+    ///
+    /// # Safety
+    ///
+    /// The read object should have the same layout as the type of val.
+    /// No checked is performed to check whether the returned object makes sense or not.
+    // TODO: Use sized array when const generics are stabilized.
+    #[expect(clippy::uninit_vec)]
+    pub unsafe fn read_mem_val<T>(&self, addr: GuestAddr) -> Result<T, QemuRWError> {
+        // let mut val_buf: [u8; size_of::<T>()] = [0; size_of::<T>()];
+
+        let mut val_buf: Vec<u8> = Vec::with_capacity(size_of::<T>());
+        unsafe {
```

Comment:
For unsafe blocks (at least if possible), it'd be good if we started adding safety comments like
```rust
// # Safety
// The vector's capacity is f the same size. The vector will be filled in `read_mem` below.
```

This will make it easier to spot bugs later and is good practice in general.